### PR TITLE
Accessibility: remove 'Edit: ' or verbs prefixes in announced names

### LINKED ISSUE #4604: A11Y: Avoid "<verb>:" prefixes in announced names
## Bug Description

- Story title text field's announced name is "Edit: Story title"
- Background color picker's announced name is "Edit: Background color"
- Page attachment text field's announced name is "Edit: Page attachment link"

## Expected Behavior

- A11Y names for input fields don't include "Edit:" or other verb prefixes

## Additional Context

- Plugin Version: 1.0.0
- Operating System: macOS 10.15.6
- Browser: Chrome 85

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by zidianlyu**
## Summary

remove 'Edit: ' or verbs prefixes in announced names.

## Relevant Technical Choices

Update text

## To-do

N/A

## User-facing changes

Announced names update

## Testing Instructions

Enable VoiceOver to listen the announced names updatee

---

Fixes #4604

### CODE REVIEW

**Review by dreamofabear — APPROVED** [review]

**Comment by swissspidy** [comment]
Note that there are a few tests that need updating as well, as they reference elements by the aria-label.

Screenshot 2020-11-12 at 19 05 07

**Comment by miina** [comment]
Some unit tests and Karma tests are still failing and need updating.

**Comment by zidianlyu** [comment]
Fixed tests.

File: assets/src/edit-story/components/form/color/colorPreview.js [hunk 1]
```diff
@@ -178,12 +178,12 @@ function ColorPreview({
   // Allow editing always in case of solid color of if color type is missing (mixed)
   const isEditable = !colorType || colorType === 'solid';
 
-  const editLabel = __('Edit', 'web-stories');
-  const inputLabel = __('Enter', 'web-stories');
+  const buttonLabel = label + __(' button', 'web-stories');
+  const inputLabel = label + __(' input', 'web-stories');
```
Comment (dreamofabear) [inline assets/src/edit-story/components/form/color/colorPreview.js]: Same issue with string concatenation and localized text. Do we actually need these suffixes? "Button" should be determined from the ARIA role, for example.

Comment (zidianlyu) [reply assets/src/edit-story/components/form/color/colorPreview.js]: Got it! Re-organized the label and its testing logic.

Comment (dreamofabear) [reply assets/src/edit-story/components/form/color/colorPreview.js]: Did you push changes?

File: assets/src/edit-story/karma/fixture/containers/common/color.js [hunk 2]
```diff
@@ -28,15 +28,15 @@ export class Color extends Container {
   }
 
   get button() {
-    return this.getByRole('button', { name: /Edit: /i });
+    return this.getByRole('button', { name: /Color/i });
```
Comment (dreamofabear) [inline assets/src/edit-story/karma/fixture/containers/common/color.js]: Do these need to be regexes or can they be strings now?

File: assets/src/edit-story/karma/fixture/containers/designPanel/link.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ export class Link extends AbstractPanel {
   }
 
   get address() {
-    return this.getByRole('textbox', { name: /Edit: Element link/i });
+    return this.getByRole('textbox', { name: /Element link/i });
```
Comment (dreamofabear) [inline assets/src/edit-story/karma/fixture/containers/designPanel/link.js]: Ditto.

File: assets/src/edit-story/components/form/color/colorPreview.js [hunk 4]
```diff
@@ -178,12 +173,10 @@ function ColorPreview({
...
   const isEditable = !colorType || colorType === 'solid';
 
-  const editLabel = __('Edit', 'web-stories');
-  const inputLabel = __('Enter', 'web-stories');
-
   const buttonProps = {
     onClick: () => setPickerOpen(true),
-    'aria-label': `${editLabel}: ${label}`,
+    'aria-label': label,
+    role: 'button',
```
Comment (dreamofabear) [inline assets/src/edit-story/components/form/color/colorPreview.js]: Is adding `role` here necessary? Is the button `role=button` by default already?

File: assets/src/edit-story/components/form/color/colorPreview.js (hunk 1 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/form/color/colorPreview.js]: Updated and pushed.

File: assets/src/edit-story/karma/fixture/containers/common/color.js (hunk 2 above)
Comment (zidianlyu) [reply assets/src/edit-story/karma/fixture/containers/common/color.js]: Yup, updated and tested.

File: assets/src/edit-story/components/form/color/colorPreview.js (hunk 4 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/form/color/colorPreview.js]: Removed and tested.

File: assets/src/edit-story/components/form/color/test/colorPreview.js [hunk 5]
```diff
@@ -33,11 +33,9 @@ jest.mock('../getPreviewStyle', () => jest.fn());
 jest.mock('../getPreviewText', () => jest.fn());
 
 function arrange(children = null) {
-  const { getByRole, getByLabelText, queryByLabelText } = renderWithTheme(
-    children
-  );
-  const button = getByLabelText(/edit/i);
-  const input = queryByLabelText(/enter/i);
+  const { getByRole, queryByLabelText } = renderWithTheme(children);
+  const button = getByRole('button');
```
Comment (dreamofabear) [inline assets/src/edit-story/components/form/color/test/colorPreview.js]: Ditto.

File: assets/src/edit-story/components/form/color/test/color.js [hunk 6]
```diff
@@ -31,10 +31,10 @@ jest.mock('../applyOpacityChange', () => jest.fn());
 
 function arrange(props = {}) {
   const onChange = jest.fn();
-  const { queryByLabelText } = renderWithTheme(
+  const { queryByLabelText, getByRole } = renderWithTheme(
     <Color label="Color" onChange={onChange} {...props} />
   );
-  const colorPreview = queryByLabelText(/Edit: Color/);
+  const colorPreview = getByRole('button');
```
Comment (dreamofabear) [inline assets/src/edit-story/components/form/color/test/color.js]: Nit: You can pass {name: 'Color'} to disambiguate with other buttons that may be added in the future.

**Review by dreamofabear — APPROVED** [review]

File: assets/src/edit-story/karma/fixture/containers/designPanel/link.js (hunk 3 above)
Comment (zidianlyu) [reply assets/src/edit-story/karma/fixture/containers/designPanel/link.js]: Done.

File: assets/src/edit-story/components/form/color/test/colorPreview.js (hunk 5 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/form/color/test/colorPreview.js]: Done.

File: assets/src/edit-story/components/form/color/test/color.js (hunk 6 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/form/color/test/color.js]: Good call, done!

**Review by swissspidy — APPROVED** [review]

**Pull request merged**